### PR TITLE
Add option for keeping the stdout for nonzero rank processing units

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Add an option to the `run_param_pyuvsim.py` script and the `run_uvsim` and `run_uvdata_uvsim` functions to allow users to keep the output from nonzero ranks for debugging purposes.

Also did some CI & testing updates now that the new version of pyradiosky is out. Biggest change is to add a new CI job that runs the tests using the minimum required versions of pyuvdata and pyradiosky to ensure those versions work. This exposed some bugs that I also fixed. I also removed some "pragma: no cover" markings for code that runs under the newest version of pyradiosky and added tests to cover some newly exposed lines.

## Motivation and Context

By default, output from nonzero ranks is surpassed (set to /dev/null). This adds an option, useful for debugging purposes,  to keep that output.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [x] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

New feature checklist:
- [x] I have added or updated the docstrings associated with my feature using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [ ] I have updated the documentation to highlight my new feature (if appropriate).
- [ ] I have added tests to cover my new feature.
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/master/CHANGELOG.md).

### CODE REVIEW

**Review by aelanman — APPROVED** [review]
Thanks for adding this, @bhazelton !

File: scripts/run_param_pyuvsim.py [hunk 1]
```diff
@@ -16,6 +16,11 @@
 parser.add_argument('paramsfile', type=str, help='Parameter yaml file.', default=None)
 parser.add_argument('--profile', type=str, help='Time profiling output file name.')
 parser.add_argument('--quiet', action='store_true', help='Suppress stdout printing.')
+parser.add_argument(
+    '--keep_nonroot_stdout',
+    action='store_true',
+    help='Do not eedirect stdout on nonzero ranks to /dev/null.'
```
Comment (mkolopanis) [inline scripts/run_param_pyuvsim.py:22]: a small typo here
```suggestion
    help='Do not redirect stdout on nonzero ranks to /dev/null.'
```

**Review by mkolopanis — COMMENTED** [review]
Looks pretty good to me. I think just two small things from this read through.

File: scripts/run_param_pyuvsim.py (hunk 1 above)
Comment (bhazelton) [reply scripts/run_param_pyuvsim.py:22]: oops, thanks!

File: .github/workflows/testsuite.yaml [hunk 2]
```diff
@@ -142,3 +142,51 @@ jobs:
...
+          conda info -a
+          conda list
+          PYVER=`python -c "import sys; print('{:d}.{:d}'.format(sys.version_info.major, sys.version_info.minor))"`
+          if [[ $PYVER != ${{ env.PYTHON }} ]]; then
+            exit 1;
+          fi
+
+      - name: Install
+        run: |
+          pip install .
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply .github/workflows/testsuite.yaml:180]: yes, good idea!

**Review by mkolopanis — APPROVED** [review]
